modules/tinymce/src/themes/silver/main/ts/ui/header/InlineHeader.ts

```diff
@@ -169,42 +171,52 @@ export const InlineHeader = (
 
     // Docking
     if (isSticky) {
-      const action = resetDocking ? Docking.reset : Docking.refresh;
-      floatContainer.on(action);
+      floatContainer.on(stickyAction);
     }
 
     // Floating toolbar
     repositionPopups();
   };
 
-  const updateMode = (updateUi: boolean = true) => {
+  const doUpdateMode = (): boolean => {
     // Skip updating the mode if the toolbar is hidden, is
     // using a fixed container or has sticky toolbars disabled
     if (useFixedToolbarContainer || !isSticky || !isVisible()) {
-      return;
+      return false;
     }
 
-    floatContainer.on((container) => {
-      const currentMode = headerBackstage.getDockingMode();
-      const newMode = calcMode(container);
-      if (newMode !== currentMode) {
-        setupMode(newMode);
-        if (updateUi) {
-          updateChromeUi(true);
+    return floatContainer.get().exists(
+      (fc) => {
+        const currentMode: 'top' | 'bottom' = headerBackstage.getDockingMode();
```

Comment:
Is there a type that could be used here? Is it even needed since I would have guesses that TS would infer the types here?